pkg/network/ebpf/c/protocols/mysql/helpers.h

```diff
@@ -0,0 +1,66 @@
+#ifndef __MYSQL_HELPERS_H
+#define __MYSQL_HELPERS_H
+
+#include "protocols/classification/common.h"
+#include "protocols/mysql/defs.h"
+#include "protocols/sql/helpers.h"
+
+// Validates the given buffer is of the format <number><delimiter> where the number is up to 2 digits.
+// The buffer cannot be just the delimiter. On error returns -1, on success the location of the next element.
+static __always_inline int is_version_component_helper(const char *buf, char delimiter) {
+    char current_char;
+#pragma unroll MAX_VERSION_COMPONENT
+    for (unsigned i = 0; i < MAX_VERSION_COMPONENT; i++) {
+        current_char = buf[i];
+        if ('0' <= current_char && current_char <= '9') {
+            continue;
+        }
+        if (current_char == delimiter && i > 0) {
+            return i+1;
+        }
+        // Any other character is not supported.
+        break;
+   }
+   return -1;
+}
+
+// Checks if the given buffer is a null terminated string that represents a version of the format <major>.<minor>.<bugfix>
+// where the major, minor and bugfix are numbers of max 2 digits each.
+static __always_inline bool is_version(const char* buf, __u32 buf_size) {
+    if (buf_size < MIN_VERSION_SIZE) {
```

Comment:
what happens if the version is 1 digits long and your buf_size is 1 (or 2)

same for checks in lines 35 and 40